src/pages/Login.tsx

```diff
@@ -1,14 +1,56 @@
 import React from "react";
 import { useState, FormEvent } from 'react';
-import {Button, Col, Container, Form, Row} from 'react-bootstrap';
-import {Link} from "react-router-dom";
+import {Alert, Button, Col, Container, Form, Row} from 'react-bootstrap';
+import {Link, useNavigate} from "react-router-dom";
+import {signinUser} from "../services/usersService.ts";
+import {User, Auth} from "../types/types.ts"
+import {useAuth} from "../context/useAuth.tsx";
 
 const Login: React.FC = () => {
     const [username, setUsername] = useState('');
     const [password, setPassword] = useState('');
+    const [error, setError] = useState<string | null>(null);
+    const [success, setSuccess] = useState<boolean>(false);
+    const navigate = useNavigate();
+
+    const {setAuth} = useAuth();
+
+    const validateFormData = () => {
+        if (!username || !password) {
```

Comment:
nit: while this fine, having real time feedback system that shows up as a small red tool tip under the field is much more user friendly than an alert on submit. This could be an improvement for the future though